uberjar/src/test/java/io/fabric8/kubernetes/clnt/UberJarTest.java

```diff
@@ -0,0 +1,211 @@
+    assertTrue(getFileInDirectory(jarExtractedDir, "io/fabric8/kubernetes/clnt/v" + majorVersion + "_" + minorVersion + "/handlers").exists());
+    File resourceHandlerServiceFile = getFileInDirectory(jarExtractedDir, "META-INF/services/io.fabric8.kubernetes.clnt.v" + majorVersion + "_" + minorVersion + ".ResourceHandler");
+    assertTrue(resourceHandlerServiceFile.exists());
+    assertTrue(containsOpenShiftKubernetesResources(resourceHandlerServiceFile, majorVersion, minorVersion));
+    assertTrue(getFileInDirectory(jarExtractedDir, "META-INF/services/io.fabric8.kubernetes.clnt.v" + majorVersion + "_" + minorVersion + ".ExtensionAdapter").exists());
+    assertTrue(getFileInDirectory(jarExtractedDir, "META-INF/services/io.fabric8.kubernetes.clnt.v" + majorVersion + "_" + minorVersion + ".ServiceToURLProvider").exists());
+  }
+
+  private boolean containsOpenShiftKubernetesResources(File resourceHandlerServiceFile, String majorVersion, String minorVersion) throws IOException {
+    List<String> lines = Files.readAllLines(resourceHandlerServiceFile.toPath());
+    final String deploymentConfigOpenShiftHandler = (majorVersion != null && minorVersion != null) ?
+      "io.fabric8.openshift.clnt.v" + majorVersion + "_" + minorVersion + ".handlers.DeploymentConfigHandler" :
+      "io.fabric8.openshift.client.handlers.DeploymentConfigHandler";
+    final String deploymentKubernetesHandler = (majorVersion != null && minorVersion != null) ?
+      "io.fabric8.kubernetes.clnt.v" + majorVersion + "_" + minorVersion + ".handlers.apps.v1.DeploymentHandler" :
+      "io.fabric8.kubernetes.client.handlers.apps.v1.DeploymentHandler";
+
+    return lines.stream().anyMatch(line -> line.contains(deploymentConfigOpenShiftHandler)) &&
+      lines.stream().anyMatch(line -> line.contains(deploymentKubernetesHandler));
+  }
+
+  private String getMajorVersion(String projectVersion) {
+    return projectVersion.split("\\.")[0];
+  }
+
+  private String getMinorVersion(String projectVersion) {
+    String[] versionParts = projectVersion.split("\\.");
+    String minorPatchVersion = projectVersion.substring(versionParts[0].length() + 1);
+    if (minorPatchVersion.contains("-")) { // SNAPSHOT VERSION
+      return minorPatchVersion.split("-")[0];
+    } else { // RELEASE VERSION
+      return versionParts[1];
+    }
+  }
+
+  private File getFileInDirectory(File parentFile, String pathToFile) {
+    pathToFile = pathToFile.replaceAll("/", File.separator);
+    return new File(parentFile, pathToFile);
+  }
+
+  private String getProjectVersion() throws IOException {
+    List<String> pomLines = readPomLines();
+    for (String line : pomLines) {
+      if (line.contains("<version>")) {
+        line = line.replaceAll("version>", "");
+        line = line.replaceAll("/", "");
+        line = line.replaceAll("<", "");
+        line = line.replaceAll(" ", "");
+        return line;
+      }
+    }
+    return null;
+  }
```

Comment:
This method is quite dangerous as it is it, you can replace by something more deterministic (e.g. unmarshal pom.xml and access specific node):
```suggestion
  private String getProjectVersion() throws Exception {
    final DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
    final DocumentBuilder builder = factory.newDocumentBuilder();
    try (final FileInputStream fis = new FileInputStream(pomFile)) {
      final Document pom = builder.parse(fis);
      final XPath xPath = XPathFactory.newInstance().newXPath();
      return xPath.compile("/project/parent/version").evaluate(pom);
    }
  }
```